From 07302f9515f5a76b5f145394f0ae758fc2997d30 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 16 Nov 2020 17:54:17 +0100
Subject: [PATCH 9/9] Revert "block: add a return value to
 set_capacity_revalidate_and_notify"

This reverts commit 7048721dee70f157bbf66890d9c590b8dfa8dfec.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/genhd.c         | 5 +----
 include/linux/genhd.h | 2 +-
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/block/genhd.c b/block/genhd.c
index c50ddbf2a..99c64641c 100644
--- a/block/genhd.c
+++ b/block/genhd.c
@@ -49,7 +49,7 @@ static void disk_release_events(struct gendisk *disk);
  * Set disk capacity and notify if the size is not currently
  * zero and will not be set to zero
  */
-bool set_capacity_revalidate_and_notify(struct gendisk *disk, sector_t size,
+void set_capacity_revalidate_and_notify(struct gendisk *disk, sector_t size,
 					bool revalidate)
 {
 	sector_t capacity = get_capacity(disk);
@@ -63,10 +63,7 @@ bool set_capacity_revalidate_and_notify(struct gendisk *disk, sector_t size,
 		char *envp[] = { "RESIZE=1", NULL };
 
 		kobject_uevent_env(&disk_to_dev(disk)->kobj, KOBJ_CHANGE, envp);
-		return true;
 	}
-
-	return false;
 }
 
 EXPORT_SYMBOL_GPL(set_capacity_revalidate_and_notify);
diff --git a/include/linux/genhd.h b/include/linux/genhd.h
index 475b286ea..4ab853461 100644
--- a/include/linux/genhd.h
+++ b/include/linux/genhd.h
@@ -315,7 +315,7 @@ static inline int get_disk_ro(struct gendisk *disk)
 extern void disk_block_events(struct gendisk *disk);
 extern void disk_unblock_events(struct gendisk *disk);
 extern void disk_flush_events(struct gendisk *disk, unsigned int mask);
-extern bool set_capacity_revalidate_and_notify(struct gendisk *disk,
+extern void set_capacity_revalidate_and_notify(struct gendisk *disk,
 			sector_t size, bool revalidate);
 extern unsigned int disk_clear_events(struct gendisk *disk, unsigned int mask);
 
-- 
2.29.2.260.ge31aba42fb

